server/models/tests/post.test.ts

```diff
@@ -0,0 +1,41 @@
+import {makeUser, makePost, dbSync} from './testHelpers'
+
+dbSync();
+
+describe('Post Model', () => {
+    describe('Post Creation', () => {
+        test("Create a basic post and check the validity of the database entry", async () => {
+            const author = await makeUser("postTester", "poster@mail.utoronto.ca");
+            expect(author).toBeDefined();
+        
+            expect(author.userName).toContain("postTester")
+        
+            const posted = await makePost(author.id, "This is a new post!", "Woooo new post new post abcdefghhi!!");
+        
+            expect(posted).toBeDefined();
+        
+            expect(posted.UserId).toEqual(author.id);
+            expect(posted.body).toEqual("Woooo new post new post abcdefghhi!!")
+            expect(posted.title).toEqual("This is a new post!")
+        
+        });
+    });
+
+    describe('Post Validation', () => {
+        test("Attempt to make posts with invalid or empty author", async() => {
+            await expect(makePost(null, "asd", "aaaaaaaaa")).rejects.toThrowError("notNull Violation"); // NULL author
+        
+            await expect(makePost("123123", "asdddd", "Woooo attempting new post abcdefghhi!!")).rejects.toThrowError("SQLITE_CONSTRAINT"); // Non-existant uid
```

Comment:
Can you re-format this bit of code?

Reply:
Sure. My prettier wasn't working for some reason, I will fix it and autoformat the code.